src/rust/engine/fs/src/store.rs

```diff
@@ -279,13 +276,30 @@ impl Store {
         expanded_digests
       })
       .and_then(move |digests| {
+        if Store::upload_is_faster_than_checking_whether_to_upload(digests.clone()) {
+          return Ok((
+            digests
+              .keys()
+              .map(|key| key.clone())
```

Comment:
`Iterator::map(|x| x.clone())` can generally be replaced with `Iterator::cloned()`: so `digests.keys().cloned().collect()`.

Reply:
This is cool! Thank you!